taqueria-utils/taqueria-utils.ts

```diff
@@ -19,19 +19,64 @@ export const decodeJson = <T>(encoded: string): Future<TaqError, T> =>
 
 export const log = <T>(message: string) => (input: T) : T => {
     console.log(message)
-    console.log(input)
     return input
 }
 
+export const logInput = <T>(message: string) => (input: T) : T => {
+    console.log(message)
+    console.log(input)
+    return input
+}
 
 export const debug = <T>(input: T) => {
     debugger
     return input
 }
 
 const mkdirFuture = (path: string): Future<TaqError, void> => attemptP(() => Deno.mkdir(path, {recursive: true}))
+export const mkdir = (path: string) : Future<TaqError, string> => pipe(path, mkdirFuture, map(() => path))
 
-export const mkdir = (path: string) : Future<TaqError, string> => pipe(path, mkdirFuture, map (() => path))
+export const ensurePathExists = (path: string) : Future<TaqError, SanitizedAbsPath> => attemptP(async () =>{
+    try {
+        await Deno.stat(path);
+        return SanitizedAbsPath.create(path);
+    } catch(_e) {
+        return Promise.reject({ kind: 'E_INVALID_PATH_DOES_NOT_EXIST', msg: 'TODO i18n message' })
```

Comment:
I've removed my use of "TODO i18n message" - I used it in so many places that when the error was thrown, I couldn't determine where the error actually occurred. I think you're trying to follow what I've done in the past here, but it turned out to be a bad idea. Perhaps we could use this instead:

```
catch (previous) {
  // TODO: i18n error message
  return Promise.reject({ kind: 'E_INVALID_PATH_DOES_NOT_EXIST', 
    msg: 'The path doesn't exist', context: path, previous })
}
```

Reply:
I'll search for any `TODO i18n message` and fix those